pkg/client/client.go

```diff
@@ -125,6 +117,42 @@ func (p ApplyClient) createOrUpdateObj(newResource runtime.Object, forceUpdate b
 	return originalGeneration != metaNewAfterUpdate.GetGeneration(), nil
 }
 
+// RetainClusterIP sets the `spec.clusterIP` value from the given 'existing' object
+// into the 'newResource' object.
+func RetainClusterIP(newResource, existing runtime.Object) error {
+	clusterIP, found, err := clusterIP(existing)
+	if err != nil {
+		return err
+	}
+	if !found {
+		// skip
+		return nil
+	}
+	switch newResource := newResource.(type) {
+	case *corev1.Service:
+		newResource.Spec.ClusterIP = clusterIP
+	case *unstructured.Unstructured:
+		if err := unstructured.SetNestedField(newResource.Object, clusterIP, "spec", "clusterIP"); err != nil {
+			return err
+		}
+	default:
+		// do nothing, object is not a service
+	}
+	return nil
+}
+
+func clusterIP(obj runtime.Object) (string, bool, error) {
+	switch obj := obj.(type) {
+	case *corev1.Service:
+		return obj.Spec.ClusterIP, (obj.Spec.ClusterIP != ""), nil
```

Comment:
the parentheses are redundant 
```suggestion
		return obj.Spec.ClusterIP, obj.Spec.ClusterIP != "", nil
```